### PR TITLE
WebGPU: Fix engine initialization

### LINKED ISSUE #13488: WebGPU: fix engine initialization
See https://forum.babylonjs.com/t/webgpuengine-mouse-events-and-pointerlock-not-working-properly/37776

Some other initializations are not performed, see the ThinEngine constructor.

### PR DESCRIPTION
**Pull request by Popov72**
Close #13488.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13541/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13541/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13541/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13541/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13541/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13541/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13541/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13541/merge#BCU1XR#0

File: packages/dev/core/src/Engines/thinEngine.ts [hunk 1]
```diff
@@ -1025,6 +1003,29 @@ export class ThinEngine {
...
+            this.hostInformation.isMobile =
+                currentUA.indexOf("Mobile") !== -1 ||
+                // Needed for iOS 13+ detection on iPad (inspired by solution from https://stackoverflow.com/questions/9038625/detect-if-device-is-ios)
+                (currentUA.indexOf("Mac") !== -1 && IsDocumentAvailable() && "ontouchend" in document);
+        };
+
+        // Set initial isMobile value
+        this._checkForMobile();
+
+        // Set up event listener to check when window is resized (used to get emulator activation to work properly)
```
Comment (RaananW) [inline packages/dev/core/src/Engines/thinEngine.ts:1023]: I assume this is the only use case for this callback? Why doesn't it work the first time it is executed?

**Review by RaananW — APPROVED** [review]
LGTM! just a small question

File: packages/dev/core/src/Engines/thinEngine.ts (hunk 1 above)
Comment (Popov72) [reply packages/dev/core/src/Engines/thinEngine.ts:1023]: Do you mean, why do we have to call it in the window resize event? If that is so, there's this comment:
``` 
Set up event listener to check when window is resized (used to get emulator activation to work properly)
``` 
Note that I didn't write this code, I just moved it in a separate method.

Comment (RaananW) [reply packages/dev/core/src/Engines/thinEngine.ts:1023]: yeah, this is what i was referring too. I saw the comment, was just wondering if it is really needed. We can check that at a later point

**Pull request merged**